server/src/main/java/com/google/speech/tools/voxetta/servlets/UtteranceUploadServlet.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.speech.tools.voxetta;
+ 
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+ 
+/** Servlet that handles Utterance Entities. */
+@WebServlet("/upload-utterance")
+public class UtteranceUploadServlet extends HttpServlet {
+ 
+  private DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+  private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+ 
+    // Get the URL of the audio file that has been uploaded to Blobstore
+    String audioKey = getUploadedFileKey(request, "audio");
+    
+    Entity utteranceEntity = new Entity("Utterance");
```

Comment:
Try to keep business logic and data base interactions isolated from the servlet.

Ideally, we have an `UtteranceService` interface, something like

```java
public class Utterance {
  // fields to hold utterance data
}

public interface UtteranceService {
  public saveUtterance(Utterance utt);
  public getUtterance(String id);
}
```

Then we can easily create concrete implementations tying to DataStore, or whatever backend we end up using.

```java
public class DataStoreUtteranceService implements UtteranceService {
  public saveUtterance(Utterance utt) {
   datastore.save.the.things();
  }
}
```

This also makes it easier to test things, as you don't have to worry about creating requests or managing servlets in the utteranceService tests, and don't need to worry about anything really except the requests and responses in the servlet tests.

Try to keep the servlets thin, especially any logic connecting to external services should be put behind an abstract interface like this.